cli/src/commands/codegen.rs

```diff
@@ -65,41 +88,71 @@ pub async fn run(opts: Opts) -> color_eyre::Result<()> {
     codegen(
         &bytes,
         opts.derives,
+        opts.attributes,
         opts.derives_for_type,
+        opts.attribute_for_type,
         opts.substitute_types,
         opts.crate_path,
         opts.no_docs,
         opts.runtime_types_only,
+        opts.no_default_derives,
     )?;
     Ok(())
 }
 
+#[derive(Clone, Debug)]
+struct OuterAttribute(syn::Attribute);
+
+impl syn::parse::Parse for OuterAttribute {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        Ok(Self(input.call(syn::Attribute::parse_outer)?[0].clone()))
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
 fn codegen(
     metadata_bytes: &[u8],
     raw_derives: Vec<String>,
+    raw_attributes: Vec<String>,
     derives_for_type: Vec<(String, String)>,
+    attribute_type: Vec<(String, String)>,
```

Comment:
```suggestion
    attributes_for_type: Vec<(String, String)>,
```

For consistency also :)

Reply:
:heavy_check_mark: